primitives/core/src/offchain/testing.rs

```diff
@@ -156,10 +158,12 @@ impl OffchainState {
 	}
 
 	fn fulfill_expected(&mut self, id: u16) {
-		if let Some(mut req) = self.expected_requests.remove(&RequestId(id)) {
-			let response = req.response.take().expect("Response checked while added.");
+		if let Some(mut req) = self.expected_requests.remove(&RequestId(self.request_counter)) {
+			let response = req.response.take().expect("Response checked when added.");
 			let headers = std::mem::take(&mut req.response_headers);
 			self.fulfill_pending_request(id, req, response, headers);
+
+			self.request_counter = self.request_counter.checked_add(1).expect("The max number of mocked requests is u16::MAX");
```

Comment:
```suggestion
			self.request_counter += 1;
```

We run tests with debug assertions enabled.